### ISSUE TITLE
using Network printer and catching socket errors

### ISSUE THREAD

**Issue by eccentric** [python-escpos/python-escpos#303]
This is very similar to issue #189, but in this case trying to catch a socket error to a network printer results in the same type of issue where the exception gets caught, but another exception happens during the garbage collection (I presume), when calling close() on the Network printer object.

Example:
```
try:
    p = printer.Network('192.168.0.100', timeout=5)
except socket.error:
    raise SystemError('error connecting to network printer')
```

If this exception is caught, another exception is raised:
```
 raise SystemError('could not connect to network printer')
File "/usr/local/lib/python2.7/dist-packages/escpos/escpos.py", line 46, in __del__
  self.close()
File "/usr/local/lib/python2.7/dist-packages/escpos/printer.py", line 225, in close
  self.device.shutdown(socket.SHUT_RDWR)
File "/usr/lib/python2.7/socket.py", line 228, in meth
  return getattr(self._sock,name)(*args)
socket.error: (107, 'Transport endpoint is not connected')
```

The socket isn't connected and so the shutdown command fails in the cleanup of the device.

This can be solved by adding a connected variable to the Network class and setting it to True only at the end of the open method of the printer.Network class, and then only calling the self.device.shutdown command if the self.connected variable is True. (there doesn't seem to be a good way to check if the self.device socket is connected or not?)

I'm not entirely sure if this is a good fix for this issue, but it solves the problem I was running into.

**Comment by patkan** [comment]
Using a variable for the socket-state seems like a viable way to me. Feel free to create a PR :-)

**Comment by belono** [comment]
@patkan this issue was also fixed at #201.
As in #189, the issue is no longer reproducible since v.3.0a so you can close as completed.
